src/operation/abort_transaction/test.rs

```diff
@@ -0,0 +1,66 @@
+use pretty_assertions::assert_eq;
```

Comment:
I'm having trouble finding a paper trail for this but I remember having a discussion a while back about moving away from these types of tests -- IIRC we agreed that in general it's more preferable to test the driver's behavior rather than the lower-level details of how operations are being constructed. To that end, I'd prefer to test this behavior with command monitoring rather than via operation building tests. It would even be fairly simple to write unified tests to do this, although that may be outside the scope of this work. WDYT @abr-egn @patrickfreed ?

Reply:
Yeah I don't think we formally discussed it, but I agree that we should probably not write any new operation unit tests, since in my experience they've been more work to maintain than they've been worth vs regular integration tests.

Regarding spec tests, it looks like they actually [already exist](https://github.com/mongodb/mongo-rust-driver/pull/513), albeit in the v2 format rather than unified, and in fact we're [already running them](https://github.com/mongodb/mongo-rust-driver/blob/master/src/test/spec/read_write_concern/operation.rs#L11), so maybe no new testing is necessary? Though I'm curious why these tests weren't failing if there are code changes in the actual driver code in this PR. Are those changes necessary? Or are there holes in the read-write-concern spec tests?

Reply:
I wrote some logic in the v2 runner to strip out empty write concerns when I first noticed this issue/filed the ticket: https://github.com/mongodb/mongo-rust-driver/blob/b843bfab8560525ae47095efab7fa7318d98243c/src/test/spec/v2_runner/operation.rs#L149
My overall comment from my review notes that we should delete this logic. I thought I remembered that this was behavior in an unrelated test, but given that there are dedicated tests for this I think removing the TODOs should be sufficient.

Reply:
Removed the unit tests I added in https://github.com/mongodb/mongo-rust-driver/pull/513/commits/1d1c8f7bb26fb46c040618b08678647cfa0ca60f.